src/sqlfluff/rules/L003.py

```diff
@@ -416,358 +466,220 @@ def _process_current_line(
+                # have a violation if that wasn't the case.
+
+                # Does the comparison line have enough unused indent to get us back
+                # to where we need to be? NB: This should only be applied if this is
+                # a CLOSING bracket.
+
+                # First work out if we have some closing brackets, and if so, how
+                # many.
+                b_idx = 0
+                b_num = 0
+                while True:
+                    if len(this_line.line_buffer[b_idx:]) == 0:
+                        break
+
+                    elem = this_line.line_buffer[b_idx]
+                    if not elem.is_code:
+                        b_idx += 1
+                        continue
+                    else:
+                        if elem.is_type("end_bracket", "end_square_bracket"):
                             b_idx += 1
+                            b_num += 1
                             continue
-                        else:
-                            if elem.is_type("end_bracket", "end_square_bracket"):
-                                b_idx += 1
-                                b_num += 1
-                                continue
-                            break  # pragma: no cover
-
-                    if b_num >= indent_diff:
-                        # It does. This line is fine.
-                        pass
-                    else:
-                        # It doesn't. That means we *should* have an indent when
-                        # compared to this line and we DON'T.
-                        memory["problem_lines"].append(this_line_no)
-                        return LintResult(
-                            anchor=trigger_segment,
-                            memory=memory,
-                            description="Indent expected and not found compared to line"
-                            " #{}".format(k),
-                            # Add in an extra bit of whitespace for the indent
-                            fixes=[
-                                LintFix.create_before(
-                                    trigger_segment,
-                                    [
-                                        WhitespaceSegment(
-                                            raw=self._make_indent(
-                                                indent_unit=self.indent_unit,
-                                                tab_space_size=self.tab_space_size,
-                                            ),
-                                        ),
-                                    ],
-                                ),
-                            ],
-                        )
-                elif this_indent_num < comp_indent_num:
-                    memory["problem_lines"].append(this_line_no)
+                        break  # pragma: no cover
+
+                if b_num < indent_diff:
+                    # It doesn't. That means we *should* have an indent when
```

Comment:
This Code is not correct at all. It takes Close brackets into account and not DEDENTs, the loop[ing mechanism is kind of crazy and if it "works" it is only covering exactly the tests related and the right outcome is basically by chance